algebras/algebra/src/test/scala/endpoints/algebra/server/SumTypedEntitiesTestSuite.scala

```diff
@@ -0,0 +1,69 @@
+package endpoints.algebra.server
+
+import akka.http.scaladsl.model.{ContentTypes, HttpMethods, HttpRequest}
+import endpoints.algebra
+
+trait SumTypedEntitiesTestSuite[
+    T <: algebra.SumTypedEntitiesTestApi
+] extends ServerTestBase[T] {
+
+  "Sum typed route" should {
+
+    "handle `text/plain` content-type requests" in {
+      serveIdentityEndpoint(serverApi.sumTypedEndpoint) { port =>
+        val request =
+          HttpRequest(HttpMethods.POST, s"http://localhost:$port/user-or-name")
+            .withEntity(ContentTypes.`text/plain(UTF-8)`, "Alice")
+        whenReady(sendAndDecodeEntityAsText(request)) {
+          case (response, entity) =>
+            assert(response.status.intValue() == 200)
+            assert(
+              response.entity.contentType == ContentTypes.`text/plain(UTF-8)`
+            )
+            entity shouldEqual "Alice"
+        }
+        ()
+      }
+    }
+
+    "handle `application/json` content-type requests" in {
+      serveIdentityEndpoint(serverApi.sumTypedEndpoint) { port =>
+        val request =
+          HttpRequest(HttpMethods.POST, s"http://localhost:$port/user-or-name")
+            .withEntity(
+              ContentTypes.`application/json`,
+              "{\"name\":\"Alice\",\"age\":42}"
+            )
+        whenReady(sendAndDecodeEntityAsText(request)) {
+          case (response, entity) =>
+            assert(response.status.intValue() == 200)
+            assert(
+              response.entity.contentType == ContentTypes.`application/json`
+            )
+            ujson.read(entity) shouldEqual ujson.Obj(
+              "name" -> ujson.Str("Alice"),
+              "age" -> ujson.Num(42)
+            )
+        }
+        ()
+      }
+    }
```

Comment:
Do you mind adding a test with a request with a content-type `application/json` but with an invalid JSON entity (e.g. omit the `age` field), and check that the error is 400, not 415?

Reply:
Good idea! That is an important thing to test.